### ISSUE TITLE
[Programmatic Access - Make code]: Nested interactive controls are not always announced by the screen readers.

### ISSUE THREAD

**Issue by Keertisree** [microsoft/pxt-microbit#5464]
**User Experience**
If nested interactive controls present on page are not defined properly making them inaccessible to screen reader, then visually impaired users who depend on assistive technology as screen reader does not announce the information of nested interactive controls properly hence AT users will not be able to perform task properly due to incomplete information of nested interactive controls.

**Note**: User credentials should NOT be included in the bug.

**Repro Steps:** 
1. Open URL: https://arcade.makecode.com/ and login with valid credentials.
2. Turn on the AI.
3. Navigate to the "View all" button with screen reader and observe the behavior.
4. Verify this issue.

### Actual Result:
Nested interactive controls are not announced by screen readers for "View all" button.
**Observation:** While navigating with screen reader, It is announcing as "My project region my project view all button view all button view all project to activate press space bar" when focus moves to 'View all' button.

### Expected Result:
Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies.

**Title**: WCAG 4.1.2: Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies (.column[role="button"])
Tags: Accessibility, WCAG 4.1.2, nested-interactive

**Issue**: Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies (nested-interactive - https://dequeuniversity.com/rules/axe/4.7/nested-interactive?application=msftAI)

**Target application**: Microsoft MakeCode Arcade - https://arcade.makecode.com/

**Element path**: .column[role="button"]

**Snippet**: <div class="column" role="button" style="z-index: 1;"><h2 class="ui header myproject-header">My Projects<span class="view-all-button" tabindex="0" title="View all projects" role="button">View All</span></h2></div>

**How to fix :** 
Fix any of the following:
  Element has focusable descendants

Environment: Chrome version 121.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.41.0 (axe-core 4.7.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.
<img width="1331" alt="MAS4 1 2-Nested interactive controls are not always announced by the screen readers" src="https://github.com/microsoft/pxt-microbit/assets/158543478/2d744df7-bcd0-4c1c-8574-dc85ceec5f1f">

**Comment by Keertisree** [comment]
#Accessibility;#A11yMAS;#A11ySev2;#E2E-HCL-MakeCode-Jan2024;#HCL;MAS4.1.2;#Win11-Chrome;#Benchmark;#A11yAuto;#K4W;

**Comment by vroopea** [comment]
As checked the issue on URL: https://makecode.microbit.org/. The issue is still repro and severity is fine. #Regressed:04-08-24;InternalReviewed-Apr2024

**Comment by srietkerk** [comment]
This is fixed. Please visit https://makecode.microbit.org/beta and https://arcade.makecode.com/beta# to see the fix.
